sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -104,5 +104,27 @@ public Response SendRequest(Request request, CancellationToken cancellationToken
             Send(message, cancellationToken);
             return message.Response;
         }
+
+        /// <summary>
+        /// Creates a scope in which all outgoing requests would use the provided
+        /// </summary>
+        /// <param name="clientRequestId">The client request id value to be sent with request.</param>
+        /// <returns>The <see cref="IDisposable"/> instance that needs to be disposed when client request id shouldn't be sent anymore.</returns>
+        /// <example>
+        /// Sample usage:
+        /// <code snippet="Snippet:ClientRequestId">
+        /// var secretClient = new SecretClient(new Uri(&quot;&lt;uri&gt;&quot;), new DefaultAzureCredential());
+        ///
+        /// using (HttpPipeline.CreateClientRequestIdScope(&quot;&lt;custom-client-request-id&gt;&quot;))
+        /// {
+        ///     // The HTTP request resulting from the client call would have x-ms-client-request-id value set to &lt;custom-client-request-id&gt;
+        ///     secretClient.GetSecret(&quot;&lt;secret-name&gt;&quot;);
+        /// }
+        /// </code>
+        /// </example>
+        public static IDisposable CreateClientRequestIdScope(string? clientRequestId)
+        {
+            return ReadClientRequestIdPolicy.StartScope(clientRequestId);
```

Comment:
Does this policy need to be added to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L35 as well?

Reply:
It's already there https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L51

Reply:
Thank you for scrolling down those 16 lines.  I clearly couldn't be bothered to do it myself.  :smile:  I don't know why, but I thought you were writing a new policy for this.  It does make more sense to reuse the existing one.